features/graph/shell/src/main/java/org/opennms/netmgt/graph/shell/GraphSearchCommand.java

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019-2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.graph.shell;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Completion;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.opennms.netmgt.graph.api.generic.GenericGraph;
+import org.opennms.netmgt.graph.api.generic.GenericVertex;
+import org.opennms.netmgt.graph.api.search.GraphSearchService;
+import org.opennms.netmgt.graph.api.search.SearchCriteria;
+import org.opennms.netmgt.graph.api.search.SearchSuggestion;
+import org.opennms.netmgt.graph.api.service.GraphService;
+
+/**
+ * Exposes the vertex search as karaf command.
+ *
+ * Log into console via: ssh -p 8101 admin@localhost
+ * Install: feature:install opennms-graph-shell
+ * Usage: type 'graph:search --namespace application --search something' in karaf console
+ * Autocomplete is supported
```

Comment:
Let's not document this here, as it will be obsolete very quickly. 
The usage will be provided by karaf anyways.

If you want to document it, put it in the admin guide instead.

Reply:
I don't believe that the admin guide will be more up to date than the documentation in the code itself - I removed it.